src/coreclr/pal/src/misc/sysinfo.cpp

```diff
@@ -431,7 +431,39 @@ GlobalMemoryStatusEx(
     // We do this only when we have the total physical memory available.
     if (lpBuffer->ullTotalPhys > 0)
     {
-#ifndef __APPLE__
+#if defined (__APPLE__)
```

Comment:
`GlobalMemoryStatusEx` in PAL is unused (outside PAL tests); good candidate of deletion:
i.e. everything this query reports: `git grep GlobalMemoryStatusEx ':/src/coreclr/pal'`

Reply:
I have submitted the cleanup in separate PR: https://github.com/dotnet/runtime/pull/91554

@Thefrank Could you please revert changes in this file? It is dead code.